Invalid generated code for variable array global variable

##### Brief Description

I would  like to generate bindings for [Gammu](https://github.com/gammu/gammu), a C library that is freely available. when trying to do so it gives an error saying it does not find the gammu.lib file nor the headers. I suspect I am not putting them at the right place. where should I put the files so that cppsharp sees them ? 

OS: Windows 
here is a sample of my generator

` class MyGenerator : ILibrary
    {


        private const string GammuRootPath = @"C:\Program Files\Gammu 1.38.1";
        private const string GammuIncludePath = GammuRootPath + @"\include\";
        private const string GammuLibPath = GammuRootPath + @"\lib\";


        public void Postprocess(Driver driver, ASTContext ctx)
        {
            //throw new NotImplementedException();
        }

        public void Preprocess(Driver driver, ASTContext ctx)
        {
            //throw new NotImplementedException();
        }

        public void Setup(Driver driver)
        {
          

            PrepareDriver(driver);

        }

        private static void PrepareDriver(Driver driver)
        {
            var drOptions = driver.Options;
            drOptions.GeneratorKind = GeneratorKind.CSharp;
            //drOptions.CompileCode = true;
            drOptions.OutputDir = "GammuSharp";
            drOptions.LibraryName = "libGammuSharp";
          
            drOptions.Libraries.Add("Gammu.lib");
            
            drOptions.OutputInteropIncludes = true;
            drOptions.GenerateInternalImports = true;
            //drOptions.Target = CompilationTarget.StaticLibrary;
            drOptions.Platform = TargetPlatform.Windows;
            drOptions.VsVersion = VisualStudioVersion.VS2012;
            drOptions.UseHeaderDirectories = true;
            
            //drOptions.OutputNamespace = "GammuSharp";
            
            // header files to include

            //drOptions.Headers.Add(GammuIncludePath + "gammu.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-wap.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-unicode.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-types.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-statemachine.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-smsd.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-settings.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-security.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-ringtone.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-nokia.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-misc.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-message.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-memory.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-limits.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-keys.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-inifile.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-info.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-file.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-error.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-debug.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-datetime.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-config.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-category.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-callback.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-call.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-calendar.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-bitmap.h");
            //drOptions.Headers.Add(GammuIncludePath + "gammu-backup.h");



        }

        public void SetupPasses(Driver driver)
        {
            
        }
    }
`

Hello @ddobrev ,
 I have been able to successfully generate bindings for the library mentioned above. 
however I have also used the   `//driver options to compile code drOptions.CompileCode = true;` 
but when it is compiling, CppSharp is failing to compile and gives 2 errors : ```

C:\Users\Kevin\documents\visual studio 2015\Projects\GammuCppSharp\GammuCppSharp.CLI\bin\Debug\GammuSharp\GammuSharp.cs(2519,15) : error CS0208: Impossible de prendre l'adresse, d'obtenir la taille ou de déclarer un pointeur vers un type managé ('GSMCodeName')
C:\Users\Kevin\documents\visual studio 2015\Projects\GammuCppSharp\GammuCppSharp.CLI\bin\Debug\GammuSharp\GammuSharp.cs(2529,15) : error CS0208: Impossible de prendre l'adresse, d'obtenir la taille ou de déclarer un pointeur vers un type managé ('GSMCodeName')
C:\Users\Kevin\documents\visual studio 2015\Projects\GammuCppSharp\GammuCppSharp.CLI\bin\Debug\GammuSharp\GammuSharp.cs(17677,37) : error CS0111: Le type 'corecrt_search' définit déjà un membre appelé 'Lfind' avec les mêmes types de paramètre
C:\Users\Kevin\documents\visual studio 2015\Projects\GammuCppSharp\GammuCppSharp.CLI\bin\Debug\GammuSharp\GammuSharp.cs(17688,37) : error CS0111: Le type 'corecrt_search' définit déjà un membre appelé 'Lsearch' avec les mêmes types de paramètre
```
any help regarding this ???

--- a/src/Generator/Types/TypeIgnoreChecker.cs
+++ b/src/Generator/Types/TypeIgnoreChecker.cs
@@ -138,13 +138,21 @@ namespace CppSharp
                 return false;
             }
 
+            var arrayElemType = array.Type.Desugar();
+
+            if (array.SizeType == ArrayType.ArraySize.Incomplete &&
+                array.QualifiedType.Qualifiers.IsConst &&
+                !arrayElemType.IsPrimitiveType())
+            {
+                Ignore();
+                return false;
+            }
+
             if (!array.QualifiedType.Visit(this))
                 return false;
 
             if (array.SizeType != ArrayType.ArraySize.Constant)
                 return true;
-
-            var arrayElemType = array.Type.Desugar();
 
             Class @class;
             if (arrayElemType.TryGetClass(out @class) && @class.IsRefType)

--- a/tests/CSharp/CSharp.h
+++ b/tests/CSharp/CSharp.h
@@ -1204,3 +1204,5 @@ int TestIndexedProperties::operator[](const int& key)
 {
     return key;
 }
+
+extern const ComplexArrayElement ArrayOfVariableSize[];

